### PR TITLE
Improve gRPC supported platforms doc

### PR DESCRIPTION
**Pull request by JamesNK**
Split the gRPC supported platform doc into two sections: one for hosting services and another for calling services.

Internal previews:
* https://review.docs.microsoft.com/en-us/aspnet/core/grpc/netstandard?view=aspnetcore-3.0&branch=pr-en-us-21689
* https://review.docs.microsoft.com/en-us/aspnet/core/grpc/supported-platforms?view=aspnetcore-3.0&branch=pr-en-us-21689

---

* ~Blocked on next Grpc.AspNetCore 2.36.0 being released.~ Done
* ~Blocked on WinHttpHandler HTTP/2 trailers being merged - https://github.com/dotnet/runtime/pull/48704~

### CODE REVIEW

File: aspnetcore/grpc/supported-platforms.md
Comment (Rick-Anderson) [inline aspnetcore/grpc/supported-platforms.md:23]:
```suggestion
* [`application/grpc-web`](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-WEB.md) - gRPC-Web modifies the gRPC protocol to be compatible with HTTP/1.1. gRPC-Web can be used in more places. gRPC-Web can be used by browser apps and in networks without complete support for HTTP/2. Two advanced gRPC features are no longer supported: client streaming and bidirectional streaming.
```

File: aspnetcore/grpc/supported-platforms.md
Comment (Rick-Anderson) [inline aspnetcore/grpc/supported-platforms.md:25]:
```suggestion
gRPC on .NET supports HTTP 1.1 and HTTP/2 , HTTP/2 is used by default. For information on setting up gRPC-Web, see <xref:grpc/browser>.
```

**Review by Rick-Anderson — APPROVED** [review]

**Comment by Rick-Anderson** [comment]
Verified all new links.

File: aspnetcore/grpc/supported-platforms.md
Comment (JamesNK) [reply aspnetcore/grpc/supported-platforms.md]: Updated change slightly. Would prefer to talk about wire format than HTTP version.

Comment (Rick-Anderson) [reply aspnetcore/grpc/supported-platforms.md]: That's fine, only because you define it above.

> HTTP/2 isn't supported everywhere, but a second wire-format using HTTP/1.1 is available for gRPC:

It's common jargon that can be difficult for non-experts. I doubt it MTs.

**Comment by JamesNK** [comment]
@Rick-Anderson I've made some substantial changes and split some sections into a new doc. Please re-review 🙏

File: aspnetcore/grpc/netstandard.md
Comment (Rick-Anderson) [inline aspnetcore/grpc/netstandard.md:57]:
```suggestion
* A reference to the [System.Net.Http.WinHttpHandler](https://www.nuget.org/packages/System.Net.Http.WinHttpHandler/) NuGet package.
```

File: aspnetcore/grpc/netstandard.md
Comment (Rick-Anderson) [inline aspnetcore/grpc/netstandard.md:74]:
```suggestion
> * The required version of `System.Net.Http.WinHttpHandler` is not currently available on NuGet.org. The latest pre-release version is available on [this NuGet feed](https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json) should be used.
```

File: aspnetcore/grpc/netstandard.md
Comment (Rick-Anderson) [inline aspnetcore/grpc/netstandard.md:78]:
```suggestion
An alternative option for .NET Framework is to use [gRPC C# core-library](https://grpc.io/docs/languages/csharp/quickstart/) to make gRPC calls. It's a third party library that supports .NET Framework and Xamarin. gRPC C-core is not supported by Microsoft.
```

**Review by Rick-Anderson — APPROVED** [review]
Suggestions

**Comment by JamesNK** [comment]
Looks good. Will wait until gRPC release before merging. Should be next week.

File: aspnetcore/grpc/supported-platforms.md
Comment (Rick-Anderson) [inline aspnetcore/grpc/supported-platforms.md:34]:
```suggestion
> * .NET Core 3.1
```

File: aspnetcore/grpc/supported-platforms.md
Comment (Rick-Anderson) [inline aspnetcore/grpc/supported-platforms.md:76]:
```suggestion
The [Grpc.Net.Client](https://www.nuget.org/packages/Grpc.Net.Client/) package supports gRPC over HTTP/2 on .NET Core 3.1 and .NET 5 or later.
```

File: aspnetcore/grpc/supported-platforms.md
Comment (Rick-Anderson) [inline aspnetcore/grpc/supported-platforms.md:83]:
```suggestion
> * .NET Core 3.1
```

**Review by Rick-Anderson — APPROVED** [review]
Should be 3.1

File: aspnetcore/grpc/supported-platforms.md
Comment (JamesNK) [reply aspnetcore/grpc/supported-platforms.md:34]: gRPC for .NET supports .NET Core 3

**Pull request merged**
